src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/CodeSnippets.cs

```diff
@@ -671,6 +671,36 @@ interface IOtherInterface
             }
             """;
 
+        public string ForwarderWithPreserveSigAndOut => $$"""
+            using System;
+            using System.Runtime.CompilerServices;
+            using System.Runtime.InteropServices;
+            using System.Runtime.InteropServices.Marshalling;
+
+            #nullable enable
+            
+            [assembly:DisableRuntimeMarshalling]
+
+            {{GeneratedComInterface()}}
+            partial interface IValue
+            {
+            }
+
+            {{GeneratedComInterface()}}
+            partial interface INativeAPIBase
+            {
+                [PreserveSig]
+                int FindValue(int key, out IValue? value);
+            }
+
+            {{GeneratedComInterface()}}
+            unsafe partial interface INativeDerived : INativeAPIBase
+            {
+                [PreserveSig]
+                int GetName(out char* name);
+            }
```

Comment:
Do we already have a test for `ref`? What about `in`?

Reply:
If not, feel free to add them to the existing signature. or a new one for kind.

Also, `ref readonly`?

Reply:
I'll refactor this test to validate all of the kinds.